### PR TITLE
SW-2684 / SW-2703 render my account page for orphaned users

### PR DESCRIPTION
**Pull request by karthikbtf**
- the layout is gross but this is a start
- i have assigned the ticket to design team as well
- with this maybe they have a working version to test with

Terraware App 2023-01-06 09-48-25

### CODE REVIEW

**Comment by constanzauanini** [comment]
Looks good! Just one thing I noticed, is that you can't go back to the home page after going to your profile. Just using the browser back button (but if you edited your profile it would be more steps to go back) or by the url. But I assume design will review this.

**Comment by karthikbtf** [comment]
> Looks good! Just one thing I noticed, is that you can't go back to the home page after going to your profile. Just using the browser back button (but if you edited your profile it would be more steps to go back) or by the url. But I assume design will review this.

good catch, i noticed that as well.. maybe we should just remove MyAccount from the menu? we don't show it on the mobile menu

**Comment by karthikbtf** [comment]
> > Looks good! Just one thing I noticed, is that you can't go back to the home page after going to your profile. Just using the browser back button (but if you edited your profile it would be more steps to go back) or by the url. But I assume design will review this.
> 
> good catch, i noticed that as well.. maybe we should just remove MyAccount from the menu? we don't show it on the mobile menu

Yes, i guess design needs to comment on this.

**Comment by hannah-cai** [comment]
> > > Looks good! Just one thing I noticed, is that you can't go back to the home page after going to your profile. Just using the browser back button (but if you edited your profile it would be more steps to go back) or by the url. But I assume design will review this.
> > 
> > 
> > good catch, i noticed that as well.. maybe we should just remove MyAccount from the menu? we don't show it on the mobile menu
> 
> Yes, i guess design needs to comment on this.

what's the issue with this ? I'm not following 😅
the visual updates look good if that's what you were looking for a review for!

**Comment by hannah-cai** [comment]
> > > > > Looks good! Just one thing I noticed, is that you can't go back to the home page after going to your profile. Just using the browser back button (but if you edited your profile it would be more steps to go back) or by the url. But I assume design will review this.
> > > > 
> > > > 
> > > > good catch, i noticed that as well.. maybe we should just remove MyAccount from the menu? we don't show it on the mobile menu
> > > 
> > > 
> > > Yes, i guess design needs to comment on this.
> > 
> > 
> > what's the issue with this ? I'm not following 😅 the visual updates look good if that's what you were looking for a review for!
> 
> Thanks @hannah-cai yes the visual updates feedback was what we were looking for.. and one more issue.. The only other issue pointed out by Constanza is that.. once user is in this MyAccount view, there's no way for user to go back to org-empty-state view. Perhaps we need a back-to link or make the 'terraformation' logo clickable (i suppose most users would expect the logo to be clickable and go-back-home).
> 
> Also fwiw - the user menus for desktop vs mobile-web seem quite different - in orphaned user view. Not sure if that was intended.. not sure who knows.

ahh I see what you mean. I think your suggestion for making the logo clickable would be a good solution!

as for the user menus, I think I had actually only specced mobile app navigation 😅 but there's no reason that it wouldn't work for mobile web as well. what's currently implemented looks different from the spec though, here's the spec for [account](https://www.figma.com/file/DftqOeGx8irCdRf7IZ4lbc/Navigation?node-id=5012%3A195776&t=oncQ2LToqVGMEfUo-11) and [the left nav](https://www.figma.com/file/DftqOeGx8irCdRf7IZ4lbc/Navigation?node-id=5001%3A209751&t=oncQ2LToqVGMEfUo-11) 

on a related note, the notification popup styling is different on desktop vs mobile web as well. the mobile web version is missing some colors.

**Comment by karthikbtf** [comment]
> > > > > > Looks good! Just one thing I noticed, is that you can't go back to the home page after going to your profile. Just using the browser back button (but if you edited your profile it would be more steps to go back) or by the url. But I assume design will review this.
> > > > > 
> > > > > 
> > > > > good catch, i noticed that as well.. maybe we should just remove MyAccount from the menu? we don't show it on the mobile menu
> > > > 
> > > > 
> > > > Yes, i guess design needs to comment on this.
> > > 
> > > 
> > > what's the issue with this ? I'm not following 😅 the visual updates look good if that's what you were looking for a review for!
> > 
> > 
> > Thanks @hannah-cai yes the visual updates feedback was what we were looking for.. and one more issue.. The only other issue pointed out by Constanza is that.. once user is in this MyAccount view, there's no way for user to go back to org-empty-state view. Perhaps we need a back-to link or make the 'terraformation' logo clickable (i suppose most users would expect the logo to be clickable and go-back-home).
> > Also fwiw - the user menus for desktop vs mobile-web seem quite different - in orphaned user view. Not sure if that was intended.. not sure who knows.
> 
> ahh I see what you mean. I think your suggestion for making the logo clickable would be a good solution!
> 
> as for the user menus, I think I had actually only specced mobile app navigation 😅 but there's no reason that it wouldn't work for mobile web as well. what's currently implemented looks different from the spec though, here's the spec for [account](https://www.figma.com/file/DftqOeGx8irCdRf7IZ4lbc/Navigation?node-id=5012%3A195776&t=oncQ2LToqVGMEfUo-11) and [the left nav](https://www.figma.com/file/DftqOeGx8irCdRf7IZ4lbc/Navigation?node-id=5001%3A209751&t=oncQ2LToqVGMEfUo-11)
> 
> on a related note, the notification popup styling is different on desktop vs mobile web as well. the mobile web version is missing some colors.

The mobile web menu is missing the menu items (my account and privacy policy). Not sure if this was intended or just a mistake. That's what I was referring to as broken.

Terraware App 2023-01-09 15-37-35
Terraware App 2023-01-09 15-37-15

There's no left-nav in this view, what were you referring to as looking different from the spec (besides the Account view.. which is broken yes) ?

**Comment by hannah-cai** [comment]
> > > > > > > Looks good! Just one thing I noticed, is that you can't go back to the home page after going to your profile. Just using the browser back button (but if you edited your profile it would be more steps to go back) or by the url. But I assume design will review this.
> > > > > > 
> > > > > > 
> > > > > > good catch, i noticed that as well.. maybe we should just remove MyAccount from the menu? we don't show it on the mobile menu
> > > > > 
> > > > > 
> > > > > Yes, i guess design needs to comment on this.
> > > > 
> > > > 
> > > > what's the issue with this ? I'm not following 😅 the visual updates look good if that's what you were looking for a review for!
> > > 
> > > 
> > > Thanks @hannah-cai yes the visual updates feedback was what we were looking for.. and one more issue.. The only other issue pointed out by Constanza is that.. once user is in this MyAccount view, there's no way for user to go back to org-empty-state view. Perhaps we need a back-to link or make the 'terraformation' logo clickable (i suppose most users would expect the logo to be clickable and go-back-home).
> > > Also fwiw - the user menus for desktop vs mobile-web seem quite different - in orphaned user view. Not sure if that was intended.. not sure who knows.
> > 
> > 
> > ahh I see what you mean. I think your suggestion for making the logo clickable would be a good solution!
> > as for the user menus, I think I had actually only specced mobile app navigation 😅 but there's no reason that it wouldn't work for mobile web as well. what's currently implemented looks different from the spec though, here's the spec for [account](https://www.figma.com/file/DftqOeGx8irCdRf7IZ4lbc/Navigation?node-id=5012%3A195776&t=oncQ2LToqVGMEfUo-11) and [the left nav](https://www.figma.com/file/DftqOeGx8irCdRf7IZ4lbc/Navigation?node-id=5001%3A209751&t=oncQ2LToqVGMEfUo-11)
> > on a related note, the notification popup styling is different on desktop vs mobile web as well. the mobile web version is missing some colors.
> 
> The mobile web menu is missing the menu items (my account and privacy policy). Not sure if this was intended or just a mistake. That's what I was referring to as broken.
> 
> Terraware App 2023-01-09 15-37-35
> 
> Terraware App 2023-01-09 15-37-15
> 
> There's no left-nav in this view, what were you referring to as looking different from the spec (besides the Account view.. which is broken yes) ?

huh, the mobile web menu is working fine for me both on staging and prod. 
image

for the left nav, I was referring to mobile web, which looks like this
image 
which is pretty visually different from how the account menu looks. in the spec it looks more consistent:
image

**Comment by karthikbtf** [comment]
> > > > > > > > Looks good! Just one thing I noticed, is that you can't go back to the home page after going to your profile. Just using the browser back button (but if you edited your profile it would be more steps to go back) or by the url. But I assume design will review this.
> > > > > > > 
> > > > > > > 
> > > > > > > good catch, i noticed that as well.. maybe we should just remove MyAccount from the menu? we don't show it on the mobile menu
> > > > > > 
> > > > > > 
> > > > > > Yes, i guess design needs to comment on this.
> > > > > 
> > > > > 
> > > > > what's the issue with this ? I'm not following 😅 the visual updates look good if that's what you were looking for a review for!
> > > > 
> > > > 
> > > > Thanks @hannah-cai yes the visual updates feedback was what we were looking for.. and one more issue.. The only other issue pointed out by Constanza is that.. once user is in this MyAccount view, there's no way for user to go back to org-empty-state view. Perhaps we need a back-to link or make the 'terraformation' logo clickable (i suppose most users would expect the logo to be clickable and go-back-home).
> > > > Also fwiw - the user menus for desktop vs mobile-web seem quite different - in orphaned user view. Not sure if that was intended.. not sure who knows.
> > > 
> > > 
> > > ahh I see what you mean. I think your suggestion for making the logo clickable would be a good solution!
> > > as for the user menus, I think I had actually only specced mobile app navigation 😅 but there's no reason that it wouldn't work for mobile web as well. what's currently implemented looks different from the spec though, here's the spec for [account](https://www.figma.com/file/DftqOeGx8irCdRf7IZ4lbc/Navigation?node-id=5012%3A195776&t=oncQ2LToqVGMEfUo-11) and [the left nav](https://www.figma.com/file/DftqOeGx8irCdRf7IZ4lbc/Navigation?node-id=5001%3A209751&t=oncQ2LToqVGMEfUo-11)
> > > on a related note, the notification popup styling is different on desktop vs mobile web as well. the mobile web version is missing some colors.
> > 
> > 
> > The mobile web menu is missing the menu items (my account and privacy policy). Not sure if this was intended or just a mistake. That's what I was referring to as broken.
> > Terraware App 2023-01-09 15-37-35
> > Terraware App 2023-01-09 15-37-15
> > There's no left-nav in this view, what were you referring to as looking different from the spec (besides the Account view.. which is broken yes) ?
> 
> huh, the mobile web menu is working fine for me both on staging and prod. image
> 
> for the left nav, I was referring to mobile web, which looks like this image which is pretty visually different from how the account menu looks. in the spec it looks more consistent: image

Ah, I understand why we are out of sync now 🤣 
This PR is specific to the orphaned user use-case, when a user has no orgs.
See:
Screen Shot 2022-12-21 at 1 52 58 PM

**Comment by karthikbtf** [comment]
updated the incorrect image

**Comment by hannah-cai** [comment]
> > > > > > > > > Looks good! Just one thing I noticed, is that you can't go back to the home page after going to your profile. Just using the browser back button (but if you edited your profile it would be more steps to go back) or by the url. But I assume design will review this.
> > > > > > > > 
> > > > > > > > 
> > > > > > > > good catch, i noticed that as well.. maybe we should just remove MyAccount from the menu? we don't show it on the mobile menu
> > > > > > > 
> > > > > > > 
> > > > > > > Yes, i guess design needs to comment on this.
> > > > > > 
> > > > > > 
> > > > > > what's the issue with this ? I'm not following 😅 the visual updates look good if that's what you were looking for a review for!
> > > > > 
> > > > > 
> > > > > Thanks @hannah-cai yes the visual updates feedback was what we were looking for.. and one more issue.. The only other issue pointed out by Constanza is that.. once user is in this MyAccount view, there's no way for user to go back to org-empty-state view. Perhaps we need a back-to link or make the 'terraformation' logo clickable (i suppose most users would expect the logo to be clickable and go-back-home).
> > > > > Also fwiw - the user menus for desktop vs mobile-web seem quite different - in orphaned user view. Not sure if that was intended.. not sure who knows.
> > > > 
> > > > 
> > > > ahh I see what you mean. I think your suggestion for making the logo clickable would be a good solution!
> > > > as for the user menus, I think I had actually only specced mobile app navigation 😅 but there's no reason that it wouldn't work for mobile web as well. what's currently implemented looks different from the spec though, here's the spec for [account](https://www.figma.com/file/DftqOeGx8irCdRf7IZ4lbc/Navigation?node-id=5012%3A195776&t=oncQ2LToqVGMEfUo-11) and [the left nav](https://www.figma.com/file/DftqOeGx8irCdRf7IZ4lbc/Navigation?node-id=5001%3A209751&t=oncQ2LToqVGMEfUo-11)
> > > > on a related note, the notification popup styling is different on desktop vs mobile web as well. the mobile web version is missing some colors.
> > > 
> > > 
> > > The mobile web menu is missing the menu items (my account and privacy policy). Not sure if this was intended or just a mistake. That's what I was referring to as broken.
> > > Terraware App 2023-01-09 15-37-35
> > > Terraware App 2023-01-09 15-37-15
> > > There's no left-nav in this view, what were you referring to as looking different from the spec (besides the Account view.. which is broken yes) ?
> > 
> > 
> > huh, the mobile web menu is working fine for me both on staging and prod. image
> > for the left nav, I was referring to mobile web, which looks like this image which is pretty visually different from how the account menu looks. in the spec it looks more consistent: image
> 
> Ah, I understand why we are out of sync now 🤣 This PR is specific to the orphaned user use-case, when a user has no orgs. See:
> 
> Terraware App 2023-01-09 15-57-36

ohhh, my bad for the miscommunication 😂 yes, it does seem to be broken then. I don't see why we wouldn't want "my account" and "privacy policy" in the mobile menu

**Comment by karthikbtf** [comment]
Sounds good @hannah-cai , Ok I will look into adding it in the mobile-web menu.
Re: the style issues...
1) notifications is missing some colors
2) ???
3) ???

**Comment by hannah-cai** [comment]
> Sounds good @hannah-cai , Ok I will look into adding it in the mobile-web menu. Re: the style issues...
> 
> 1. notifications is missing some colors
> 2. ???
> 3. ???

2. the mobile web left nav style should be styled like [this](https://www.figma.com/file/DftqOeGx8irCdRf7IZ4lbc/Navigation?node-id=5001%3A209751&t=oncQ2LToqVGMEfUo-11) (the tabs should be the same, but the style should be different from normal desktop view)
3. the mobile web account menu should be styled like [this](https://www.figma.com/file/DftqOeGx8irCdRf7IZ4lbc/Navigation?node-id=5012%3A195776&t=oncQ2LToqVGMEfUo-11)

**Comment by karthikbtf** [comment]
> > Sounds good @hannah-cai , Ok I will look into adding it in the mobile-web menu. Re: the style issues...
> > 
> > 1. notifications is missing some colors
> > 2. ???
> > 3. ???
> 
> 2. the mobile web left nav style should be styled like [this](https://www.figma.com/file/DftqOeGx8irCdRf7IZ4lbc/Navigation?node-id=5001%3A209751&t=oncQ2LToqVGMEfUo-11) (the tabs should be the same, but the style should be different from normal desktop view)
> 3. the mobile web account menu should be styled like [this](https://www.figma.com/file/DftqOeGx8irCdRf7IZ4lbc/Navigation?node-id=5012%3A195776&t=oncQ2LToqVGMEfUo-11)

Thanks @hannah-cai I will capture these in separate tickets - as the scope of this PR is making MyAccount work in the orphaned user view. Currently it is broken (clicking on MyAccount does nothing).

**Comment by karthikbtf** [comment]
@constanzauanini this one is good to go.
1) added fix for SW-2703 as well (discard edits on cancel)
2) added opt-in flag check for time zone
3) made logo clickable
4) fixed my account style in no-org view
5) enable some menu items in mobile web view (my account, opt-in etc.)

Terraware App 2023-01-10 10-36-45
Terraware App 2023-01-10 10-36-15
Terraware App 2023-01-10 10-35-23

File: src/components/TopBar/TopBarContent.tsx [hunk 1]
```diff
@@ -90,7 +99,12 @@ export default function TopBarContent(props: TopBarProps): JSX.Element | null {
         )}
       </Grid>
 
-      <Grid item xs={6} className={`${classes.center} logo`} />
+      <Grid
+        item
+        xs={6}
+        className={`${classes.center} ${classes.clickableLogo} logo`}
```
Comment (constanzauanini) [inline src/components/TopBar/TopBarContent.tsx:105]: Can we add cursor: pointer here? In desktop the <Link> is already adding it

Comment (karthikbtf) [reply src/components/TopBar/TopBarContent.tsx:105]: for mobile will it matter? i will add it. just curious

Comment (constanzauanini) [reply src/components/TopBar/TopBarContent.tsx:106]: Oh, you are right, not needed. I was testing with the inspector

**Review by constanzauanini — APPROVED** [review]

**Pull request merged**